services/web/server/src/simcore_service_webserver/diagnostics_monitoring.py

```diff
@@ -23,22 +23,34 @@
 kCANCEL_COUNT = f"{__name__}.cancel_count"
 
 kCOLLECTOR_REGISTRY = f"{__name__}.collector_registry"
+kPROCESS_POOL_EXECUTOR = f"{__name__}.process_pool_executor.pool"
+
+
+def get_collector_registry(app: web.Application) -> CollectorRegistry:
+    return app[kCOLLECTOR_REGISTRY]
 
 
 async def metrics_handler(request: web.Request):
-    # TODO: prometheus_client.generate_latest blocking! -> Consider https://github.com/claws/aioprometheus
-    reg = request.app[kCOLLECTOR_REGISTRY]
-    resp = web.Response(body=prometheus_client.generate_latest(registry=reg))
-    resp.content_type = CONTENT_TYPE_LATEST
-    return resp
+    registry = get_collector_registry(request.app)
+
+    with concurrent.futures.ThreadPoolExecutor() as pool:
+        # NOTE: Cannot use ProcessPoolExecutor because registry is not pickable
+        result = await request.loop.run_in_executor(
+            pool, prometheus_client.generate_latest, registry
+        )
+        response = web.Response(body=result)
+        response.content_type = CONTENT_TYPE_LATEST
+        return response
 
 
 def middleware_factory(app_name: str) -> Coroutine:
     @web.middleware
-    async def _middleware_handler(request: web.Request, handler: Coroutine):
+    async def _middleware_handler(request: web.Request, handler: Callable):
         if request.rel_url.path == "/socket.io/":
             return await handler(request)
 
+        log_exception = None
```

Comment:
what is this for?

Reply:
the variable is used in the ``finally`` section and had to initialized because there was a risk of it being unbound

Reply:
the variable is used in the ``finally`` section and had to initialized because there was a risk of it being unbound (the issue was flagged by pylance)